runtime/src/accounts_db.rs

```diff
@@ -1525,27 +1529,69 @@ impl AccountsDB {
         hasher.result()
     }
 
-    fn accumulate_account_hashes(mut hashes: Vec<(Pubkey, Hash)>) -> Hash {
-        let mut sort = Measure::start("sort");
+    fn accumulate_account_hashes(hashes: Vec<(Pubkey, Hash, u64)>) -> Hash {
+        let (hash, ..) = Self::do_accumulate_account_hashes_and_capitalization(hashes, false);
+        hash
+    }
+
+    fn accumulate_account_hashes_and_capitalization(
+        hashes: Vec<(Pubkey, Hash, u64)>,
+    ) -> (Hash, u64) {
+        let (hash, cap) = Self::do_accumulate_account_hashes_and_capitalization(hashes, true);
+        (hash, cap.unwrap())
+    }
+
+    fn do_accumulate_account_hashes_and_capitalization(
+        mut hashes: Vec<(Pubkey, Hash, u64)>,
+        calculate_cap: bool,
+    ) -> (Hash, Option<u64>) {
+        let mut sort_time = Measure::start("sort");
         hashes.par_sort_by(|a, b| a.0.cmp(&b.0));
-        sort.stop();
-        let mut hash_time = Measure::start("hash");
+        sort_time.stop();
 
-        let fanout = 16;
+        let mut sum_time = Measure::start("cap");
+        let cap = if calculate_cap {
+            Some(hashes.iter().fold(0, |acuum: u64, (_, _, lamports)| {
+                acuum.checked_add(*lamports).unwrap()
```

Comment:
I have yet to check perf comparison of checked u64 addition vs unchecked u128 addition....

Also, consider par_iter() + fold()?

Reply:
Well, I played with `godbolt` a bit: https://rust.godbolt.org/z/h47hTT

The general observation is that u128 _will be_ faster than checked u64 because it unrolls this loop. But it seems that this is currently only realized for nightly rustc... Stable rusts only manages to unroll for simple data shape like `&[u64]`. nightly even can handle a tuple like this. Our compiler is getting smarter. :)

Because checked u64 math introduces arbitrary abortion execution path at each addition step, it can't unroll.
So, I'll rewrite to use u128.